catalog/plugin/catalog-plugin-defaultsecurityattributevalues/src/main/java/org/codice/alliance/catalog/plugin/defaultsecurity/DefaultSecurityAttributeValuesPlugin.java

```diff
@@ -121,62 +130,53 @@ public Metacard addDefaults(Metacard metacard) {
 
-    final Metacard extendedMetacard;
-    MetacardImpl metacardImpl;
+    final MetacardImpl extendedMetacard;
+    // If the metacard doesn't have any of the securityAttributes AttributeDescriptors, add ALL of
+    // the securityAttributes AttributeDescriptors to the metacard.
     if (securityAttributes
         .getAttributeDescriptors()
         .stream()
         .noneMatch(ad -> metacard.getMetacardType().getAttributeDescriptors().contains(ad))) {
-      metacardImpl =
+      extendedMetacard =
           new MetacardImpl(
               metacard,
               new MetacardTypeImpl(
                   metacard.getMetacardType().getName(),
                   metacard.getMetacardType(),
                   securityAttributes.getAttributeDescriptors()));
     } else {
-      metacardImpl = new MetacardImpl(metacard);
+      extendedMetacard = new MetacardImpl(metacard);
     }
-    Set<String> updatedTags = new HashSet<>(metacard.getTags());
-    updatedTags.add(DEFAULTMARKINGS);
-    metacardImpl.setTags(updatedTags);
-    extendedMetacard = metacardImpl;
-
-    securityMarkings
-        .keySet()
-        .stream()
-        .filter(
-            securityMarking ->
-                extendedMetacard.getMetacardType().getAttributeDescriptor(securityMarking) != null)
-        .forEach(
-            securityMarking -> {
-              extendedMetacard.setAttribute(securityMarkings.get(securityMarking));
-            });
-    return extendedMetacard;
-  }
-
-  public void setClassification(String classification) {
-    addMapping(classification, Security.CLASSIFICATION);
-  }
-
-  public void setReleasability(String releasability) {
-    addMapping(releasability, Security.RELEASABILITY);
-  }
-
-  public void setCodewords(String codewords) {
-    addMapping(codewords, Security.CODEWORDS);
-  }
 
-  public void setDisseminationControls(String disseminationControls) {
-    addMapping(disseminationControls, Security.DISSEMINATION_CONTROLS);
-  }
-
-  public void setOtherDisseminationControls(String otherDisseminationControls) {
-    addMapping(otherDisseminationControls, Security.OTHER_DISSEMINATION_CONTROLS);
-  }
+    final Set<String> updatedTags = new HashSet<>(metacard.getTags());
+    for (Map.Entry<String, String> entry :
+        metacardAttributeNameToSystemHighAttributeNameMap.entrySet()) {
+      final String metacardAttributeName = entry.getKey();
+      final String systemHighAttributeName = entry.getValue();
+
+      if (extendedMetacard.getMetacardType().getAttributeDescriptor(metacardAttributeName)
+          != null) {
+        final Optional<List<Serializable>> systemHighAttributeValue =
+            systemHighAttributes.getValues(systemHighAttributeName);
+        if (systemHighAttributeValue.isPresent()) {
+          extendedMetacard.setAttribute(
+              new AttributeImpl(metacardAttributeName, systemHighAttributeValue.get()));
+          updatedTags.add(DEFAULT_MARKINGS_TAG);
+        } else {
+          LOGGER.trace(
```

Comment:
✏️  I think we can upgrade this to debug. If I was trying to troubleshoot this in the field, I would rarely consider using trace. It's too much info to sort through.